recipes/libtribler/recipe.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+DEPS_libtribler=(kivy python openssl m2crypto twisted sqlite3 pyasn1 apsw cherrypy netifaces libtorrent libnacl libsodium pil plyvel requests)
+BUILD_libtribler=$BUILD_PATH/libtribler/tribler-git
+RECIPE_libtribler=$RECIPES_PATH/libtribler
+
+function prebuild_libtribler() {
+
+	# Clone repo so all submodules are included:
+	cd $BUILD_PATH/libtribler
+	git clone --recursive https://github.com/tribler/tribler.git tribler-git
```

Comment:
Tribler uses git submodules for Dispersy and pymdht. It then makes imports such as Tribler.dispersy.etc. and something similar for pydmht. So it expects these two modules to be present inside libtribler (and not externally). Using the direct archive from GitHub means that these submodules will be missing, like Darsestheus stated.
The approach in my latest commit is that libtribler is recursively cloned (as to include submodules) to $PACKAGES_PATH/libtribler. This is only done if it hasn't been cloned before. If it has been cloned before than a git pull command is executed instead to fetch the latest changes. In case any changes were fetched (it checks this by comparing the latest commit hash before and after pulling), or in the case that the build directory is empty, the libtribler directory (in $PACKAGES_PATH/libtribler) is copied to the build directory (in $BUILD_libtribler).